course_info/icommons.py

```diff
@@ -122,7 +121,7 @@ def _course_info_instructor_list(self, course_instance_id, **kwargs):
                 instructors = []
                 for person in rv['results']:
                     course_person_schema(person)
-                    if person.get('role')['role_id'] in INSTRUCTOR_ROLE_IDS:
+                    if person.get('role') and person.get('role')['role_id'] in INSTRUCTOR_ROLE_IDS:
```

Comment:
The only way that line throws that error is if there's a `None` in `course_instructor_list`.

(also the parens around `x.get('role', {})` are unnecessary)